pages/studentPortal/index.tsx

```diff
@@ -65,6 +72,37 @@ export default function StudentPortalPage() {
     });
   }, [user]);
 
+  const [updateLastSeenModal] = useMutation(UPSERT_LAST_SEEN_MODAL);
+  const [showModal, setShowModal] = useState(false);
+
+  useQuery<FetchModalData>(FETCH_LAST_SEEN_MODAL, {
+    variables: {
+      _id: user.uid,
+    },
+
+    onCompleted: (data) => {
+      if (data.users[0].freemium_user) {
+        const lastSeenValue = data.users[0].freemium_user?.lastSeenModal;
+        const lastSeenDifference = differenceInSeconds(
+          new Date(),
+          new Date(lastSeenValue)
+        );
+
+        if (lastSeenDifference > 60) {
+          setShowModal(true);
+          updateLastSeenModal({
+            variables: { userId: user.uid, lastSeenModal: new Date() },
+          });
+        }
+      } else {
+        setShowModal(true);
```

Comment:
is there a way not to repeat yourself with this else condition. this may not be possible, but see if it becomes easier once you are querying from the freemium_users table instead of the users table.

Reply:
yes.  figured it out.